### ISSUE TITLE
[Bug]: skill scraper leaves unsloth's Quick Reference empty and two references/ files unlisted

### ISSUE THREAD

**Issue by daragao3** [NousResearch/hermes-agent#106780]
### Summary

The documentation-scraper that generates skills leaves two defects in
`optional-skills/mlops/training/unsloth`, both live on `main` today. Neither is
caught by `tests/skills/test_authoring_standards.py`, which checks frontmatter,
size and `related_skills` but not these.

### 1. Quick Reference is an unfilled placeholder — zero patterns extracted

`optional-skills/mlops/training/unsloth/SKILL.md` still carries the scraper's
placeholder instead of any pattern:

```
## Quick Reference

### Common Patterns

*Quick reference patterns will be added as you use the skill.*
```

```console
$ git show main:optional-skills/mlops/training/unsloth/SKILL.md | grep -c '^\*\*Pattern '
0
```

For contrast, the sibling skill from the same scraper does it right —
`optional-skills/mlops/training/axolotl` has 7 patterns, each a short sentence
plus a snippet.

This looks like the *same* pattern extractor that failed in the opposite
direction on `pytorch-fsdp`, where whole documentation pages were inlined as
pattern text until the 2026-08-08 standards sweep (1c9433897c / 55982159dd)
moved them to `references/common-patterns.md`. Same component, two failure
modes: everything, or nothing.

### 2. `references/` files that are listed nowhere

The directory holds four files, but both `SKILL.md` and `references/index.md`
name only `llms-txt.md`:

```console
$ git ls-tree --name-only main optional-skills/mlops/training/unsloth/references/
index.md  llms-full.md  llms-txt.md  llms.md

$ git show main:optional-skills/mlops/training/unsloth/SKILL.md | grep -oP '(?<=- \*\*)[^*]+\.md'
llms-txt.md
```

So `llms-full.md` (1,094,024 bytes) and `llms.md` (12,586) are unreachable for
an agent reading the skill — it has no way to learn they exist.

They are not redundant copies. `llms-full.md` carries **135 headings that
`llms-txt.md` does not**, including "understanding reward functions", "how grpo
trains a model", "base models" and "epochs". `llms.md` is the upstream
`llms.txt` link index.

### Suggested checks

Two mechanical checks would close both, and fit naturally alongside the
existing authoring-standards suite (and overlap with the linter proposed in
#37352):

1. No `SKILL.md` retains the scraper's placeholder string.
2. For scraper-generated skills, every `references/*.md` on disk appears in
   both the SKILL.md reference list and `references/index.md`.

A caveat on check 2 from trying it: restrict it to scraper-generated skills
(those carrying `automatically generated from official documentation` **and**
having a `references/index.md`). Run across all skills it produced 77 false
positives, because the `- **name.md** - description` bullet form is the
scraper's own and hand-written skills reference their files as prose or links.

A third check worth considering, on **max line length** rather than file size:
the `pytorch-fsdp` failure was a 130,541-character single line, and a
size-based cap alone would flag `research-paper-writing` (106,051 bytes) which
is perfectly healthy at 2,377 lines with a 334-char longest line.

### Environment

Observed on `main` @ 167448edca. Both items verified with the `git show` /
`git ls-tree` commands above against that revision, so no local checkout state
is involved.
